### PR TITLE
feat: Support Rollback multi-source apps; Follow-up to PR 14124

### PR DESCRIPTION
**Pull request by keithchong**
This is the follow-on PR for https://github.com/argoproj/argo-cd/pull/14124.  There is no Issue opened for this since these are some of the UI changes that should be part of 14124.

This also includes the fix for https://github.com/argoproj/argo-cd/pull/18200 for 2.11, but for master.

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).
* [ ] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [ ] I have added a brief description of why this PR is necessary and/or what this PR solves.
* [ ] Optional. My organization is added to USERS.md.
* [ ] Optional. For bug fixes, I've indicated what older releases this fix should be cherry-picked into (this may or may not happen depending on risk/complexity).

### CODE REVIEW

**Review by ishitasequeira — APPROVED** [review]
Thanks @keithchong !! Tested the changes and it looks good. Overall code looks good.

File: ui/src/app/applications/components/application-details/application-details.tsx [hunk 1]
```diff
@@ -180,6 +180,216 @@ export class ApplicationDetails extends React.Component<RouteComponentProps<{app
...
+                            <div className='white-box' style={{marginTop: '1.5em'}}>
+                                {sourceHeader && sourceHeader}
+                                <div className='white-box__details'>
+                                    <div className='row white-box__details-row'>
+                                        <div className='columns small-3'>SHA:</div>
+                                        <div className='columns small-9'>
+                                            <Revision repoUrl={aSource.repoURL} revision={aRevision} />
+                                        </div>
+                                    </div>
+                                </div>
```
Comment (ishitasequeira) [inline ui/src/app/applications/components/application-details/application-details.tsx:303]: @keithchong , this block looks quite redundant and used multiple times here. Do you think we could create a separate function and pass in the values that need to be displayed to it?

Comment (keithchong) [reply ui/src/app/applications/components/application-details/application-details.tsx]: Thanks @ishitasequeira , I've commoned up two sections.  BTW, the panels launched from the "..."  (f.k.a. More) are impacted, for both multi-source and single source apps

**Review by ishitasequeira — APPROVED** [review]
LGTM!!

**Pull request merged**